nav2_behavior_tree/include/nav2_behavior_tree/bt_action_node.hpp

```diff
@@ -174,7 +176,25 @@ class BtActionNode : public BT::ActionNodeBase
       // user defined callback
       on_tick();
 
-      on_new_goal_received();
+      send_new_goal();
+    }
+
+    // if new goal was sent and action server has not yet responded
+    // check the future goal handle
+    if (goal_sent_) {
```

Comment:
Could `goal_sent_` be replaced by a check on `goal_handle_` instead? If its `reset()` after each goal, then it should be a nullpointer while the goal request is being processed without needing to explicitly track this state variable. 

We'd need to make sure in `halt` and all other terminal conditions that the `goal_handle_` is `reset()` so that its only valid when no goal is sent yet or its being sent currently.